frontend/src/ClassView/ClassView.tsx

```diff
@@ -16,22 +18,86 @@ function ClassView() {
   const nav = useNavigate();
   const mode: "student" | "faculty" = "student"; // should be taken from some app state / login info
   const [cookies, setCookies, removeCookies] = useCookies(["jwt"]);
+  const [error, setError] = useState("");
+  const params = useParams();
+  const [classID, setClassID] = useState("");
+  const [assignments, setAssignments] = useState(new Array<any>());
+  getClassInfo(cookies.jwt);
+
   function handleRemoveCookies() {
     removeCookies("jwt");
     nav("/");
   }
   const navToClassStats = () => {
-    nav("/class/classstats");
+    fetchClassInfo(cookies.jwt);
+    nav("/class/" + classID + "/classstats");
```

Comment:
Please refer to the other comment in this file. Making a single request on component-mount will save a lot of trouble.

```suggestion
    nav("/class/" + classID + "/classstats");
```